pkg/api/types.go

```diff
@@ -503,6 +504,7 @@ type AgentPoolProfile struct {
 	PreprovisionExtension               *Extension           `json:"preProvisionExtension"`
 	Extensions                          []Extension          `json:"extensions"`
 	KubernetesConfig                    *KubernetesConfig    `json:"kubernetesConfig,omitempty"`
+	KubernetesVersion                   string               `json:"kubernetesVersion"`
```

Comment:
why do we need this? do we expect heterogeneous k8s version support across agentpools?

Reply:
Just to be consistent with the pre-existing types I have a slight preference for `OrchestratorVersion` here.